src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -980,38 +968,31 @@ void CMitsubaLoader::cacheEmissionProfile(SContext& ctx, const CElementEmissionP
 {
 	if (!profile)
 		return;
-	
+
 	const auto cacheKey = ctx.emissionProfileCacheKey(*profile);
 
 	const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+
 	if (ctx.override_->findCachedAsset(cacheKey, types, ctx.inner, 0u).getContents().empty())
 	{
 		auto assetLoaded = interm_getAssetInHierarchy(m_assetMgr, profile->filename, ctx.inner.params, 0u, ctx.override_);
```

Comment:
due to not needing to cache under a special key: https://github.com/Devsh-Graphics-Programming/Nabla/pull/630/files/faf4f805df55fe3af3e1a80188494c324e015919..cb61c15d7f6695c3e22f3270d9173d72f401326e#r1456673824

just use the `interm_getAssetInHierarchy` with the filename, only do special caching for the sampler cause you create that on-demand

Reply:
I tried to use interm_getAssetInHierarchy inside the IR frontend but I couldn't access m_asserManager from there. Was that intended design or can interm_getAssetInHierarchy be accessed other way?

Reply:
`interm_getAssetInHierarchy` is only accesible and usable by other IAssetLoaders (friendship)